shared/java/org/jetbrains/jwm/Clipboard.java

```diff
@@ -0,0 +1,58 @@
+package org.jetbrains.jwm;
+
+import lombok.*;
+import java.util.*;
+import java.util.function.*;
+import java.util.concurrent.*;
+import org.jetbrains.annotations.*;
+import org.jetbrains.jwm.impl.*;
+
+public class Clipboard {
+    @ApiStatus.Internal
+    public static Map<String, ClipboardFormat> _formats = Collections.synchronizedMap(new HashMap<String, ClipboardFormat>());
+
+    public static void set(ClipboardEntry ... entries) {
+        assert entries.length > 0;
+        _nSet(entries);
+    }
+
+    @Nullable
+    public static ClipboardEntry get(ClipboardFormat ... formats) {
+        assert formats.length > 0;
+        return _nGet(formats);
+    }
+
+    public static ClipboardFormat[] getFormats() {
+        return _nGetFormats();
+    }
+
+    public static void clear() {
+        _nClear();
+    }
+
+    @NotNull @SneakyThrows
+    public static ClipboardFormat registerFormat(String formatId) {
+        ClipboardFormat format = _formats.get(formatId);
+
+        if (format != null)
+            return format;
+
+        if (!_nRegisterFormat(formatId))
+            throw new RuntimeException("Failed to register format: " + formatId);
+
+        return registerFormatInternal(formatId);
```

Comment:
Need some lock between checking if format exists and actually registering new format

Reply:
I thought, our api calls are allowed only on UI thread?